Feature/design system

Ahmedsayed0895 opened the pull request.

Ahmedsayed0895 committed:
text-style-theme (#1)

* Add `TudeeTextStyle` data class, which defines text styles for headlines, titles, bodies, and labels.

* `SizedTextStyle` data class
This data class represents a set of text styles with different sizes: large, medium, and small.

* feat: Add nunito font family to the project resources.

* adding a font family with normal, medium, and semi-bold weights.
It also includes a local composition for the default text style.

* Define `defaultTextStyle` for the Tudee app.

* Refactor: Delete Typography.kt

* Define TudeeTheme composable function

This function provides the default text styles for the application.

* add textStyle to
The `AppTheme` now only provides access to `textStyle` via `LocalTudeeTextStyle`.

* Modify MainActivity to utilize TudeeTheme

* Create `themeLocals.kt` and move `LocalTudeeTextStyle` into it.

This change aims to improve code organization by separating theme-related local compositions.
The Android Gradle Plugin version has also been updated from 8.10.0 to 8.8.0.

Ahmedsayed0895 committed:
Add Nunito fonts and update .gitignore

Ahmedsayed0895 committed:
Merge remote-tracking branch 'origin/feature/design-system' into feature/design-system

saraashraf11 committed:
color-theme (#2)

* feat: Add color theme comment

* feat: add color theme comment

* add: implement light and dark themes

* refactor: update themeLocals

* refactor: add LocalTudeeColors providers

* refactor: update AppTheme

* refactor: remove unused `staticCompositionLocalOf` import

MostafaMohamed2002 commented on gradle/libs.versions.toml:
```diff
@@ -1,5 +1,5 @@
 [versions]
-agp = "8.10.0"
+agp = "8.8.0"
```
upgrade to agp = "8.10.0"

MostafaMohamed2002 commented on app/src/main/java/com/washingtondcsquad/tudee/presentation/design/theme/AppTheme.kt:
```diff
@@ -0,0 +1,24 @@
+        @ReadOnlyComposable
+        get()= LocalTudeeTextStyle.current
+
+    val tudeeColors: TudeeColors
```
i think we can change the AppTheme to TudeeTheme ,
 tudeeColors  to colors so it  can be used like TudeeTheme.colors

saraashraf11 replied:
ok I will change it

saraashraf11 replied:
I changed tudeeColors to colors, but I can't change it in AppTheme because there is another function called TudeeTheme.

MostafaMohamed2002 commented on app/src/main/java/com/washingtondcsquad/tudee/presentation/design/theme/themeLocals.kt:
we can move them to tudee/presentation/design/theme/TudeeTheme.kt

saraashraf11 replied:
ok

MostafaMohamed2002 reviewed (dismissed):
and there is formating problem , use can you this shortcut `ctrl+alt+o`

MostafaMohamed2002 commented:
<strike> also remove unused files </strike>

MostafaMohamed2002 commented:
i think it's okay to seperate design system package

kareem-01 commented on app/src/main/java/com/washingtondcsquad/tudee/presentation/design/theme/AppTheme.kt:
```diff
@@ -0,0 +1,24 @@
+import com.washingtondcsquad.tudee.presentation.design.textStyle.TudeeTextStyle
+
+@Stable
+object AppTheme {
```
اقترح Theme بس ك اسم

Ahmedsayed0895 committed:
Remove: `Colors`, `Shapes`, `Sizes`, and `Spacer` files

These files were removed as part of a code refactoring.
Update Android Gradle Plugin version to 8.10.0.

kareem-01 commented on app/src/main/java/com/washingtondcsquad/tudee/presentation/design/colors/DarkTudeeColors.kt:
```diff
@@ -0,0 +1,33 @@
+
+import androidx.compose.ui.graphics.Color
+
+val DarkTudeeColors = TudeeColors(
```
الالوان دي مش زي الديزاين 

primaryGradient
pinkAccent
yellowAccent
greenAccent
purpleAccent
error
yellowVariant
greenVariant
purpleVariant
errorVariant

saraashraf11 replied:
yes I change it , thanks

kareem-01 commented on app/src/main/java/com/washingtondcsquad/tudee/presentation/design/colors/LightTudeeColors.kt:
```diff
@@ -0,0 +1,33 @@
+
+import androidx.compose.ui.graphics.Color
+
+val LightTudeeColors = TudeeColors(
```
الاسم بادأ ب capital اول حرف هنا والدارك يتغير بس

saraashraf11 committed:
color-theme (#4)

* feat: Add color theme comment

* feat: add color theme comment

* add: implement light and dark themes

* refactor: update themeLocals

* refactor: add LocalTudeeColors providers

* refactor: update AppTheme

* refactor: remove unused `staticCompositionLocalOf` import

* refactor: consolidate theme locals and rename themeColors

* refactor: delete themeLocals

saraashraf11 committed:
origin/color-theme (#5)

* feat: Add color theme comment

* feat: add color theme comment

* add: implement light and dark themes

* refactor: update themeLocals

* refactor: add LocalTudeeColors providers

* refactor: update AppTheme

* refactor: remove unused `staticCompositionLocalOf` import

* refactor: consolidate theme locals and rename themeColors

* refactor: delete themeLocals

* refactor: update dark theme colors

* refactor: rename TudeeColors files and variables

NadaMansour20 reviewed (approved):
Great work

Mohamed-Boumaraf reviewed (approved):
Good job
